### DISCUSSION TITLE
ERROR: Unrecognized pair style 'deepmd'

### DISCUSSION THREAD

**Discussion by manbujin** [deepmodeling/deepmd-kit#2686] [Q&A]
conda create -n deepmd deepmd-kit=*=*gpu libdeepmd=*=*gpu lammps cudatoolkit=11.6 horovod -c https://conda.deepmodeling.com -c defaults
  I installed deepmd-kit and lammps with the above commands, but when I explored with dpgen, it didn't work. 
  Before running the task, the deepmd environment has been loaded.     "conda activate deepmd"
  The error message in the model_devi.log file is as follows, it says no dpplugin.so ,then I find that the envs/deepmd/lib/deepmd_lmp folder is empty and there is no dpplugin.so. 
  How can I solve this problem? Can anyone help me?

LAMMPS (23 Jun 2022 - Update 4)
OMP_NUM_THREADS environment is not set. Defaulting to 1 thread. (src/comm.cpp:98)
  using 1 OpenMP thread(s) per MPI task
Open of file /root/data/anaconda3/envs/deepmd/lib/deepmd_lmp/dpplugin.so failed: /root/data/anaconda3/envs/deepmd/lib/deepmd_lmp/dpplugin.so: undefined symbol: _ZN9LAMMPS_NS5Error4_allERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEiN3fmt6v8_lmp17basic_string_viewIcEENSA_17basic_format_argsINSA_20basic_format_contextINSA_8appenderEcEEEE
Loaded 0 plugins from /root/data/anaconda3/envs/deepmd/lib/deepmd_lmp
Reading data file ...
  triclinic box = (0 0 0) to (12.994126 11.459059 11.138661) with tilt (-6.1264935 -3.4409904 -5.6967694)
  1 by 1 by 1 MPI processor grid
  reading atoms ...
  150 atoms
  read_data CPU = 0.000 seconds
Changing box ...
  triclinic box = (0 0 0) to (12.994126 11.459059 11.138661) with tilt (-6.1264935 -3.4409904 -5.6967694)
ERROR: Unrecognized pair style 'deepmd' (src/force.cpp:279)
Last command: pair_style      deepmd ../graph.002.pb ../graph.001.pb ../graph.000.pb ../graph.003.pb  out_freq ${THERMO_FREQ} out_file model_devi.out

When I run lmp-h, the following appears
Large-scale Atomic/Molecular Massively Parallel Simulator - 23 Jun 2022 - Update 4

Usage example: /root/data/anaconda3/envs/deepmd/bin/_lmp -var t 300 -echo screen -in in.alloy

List of command line options supported by this LAMMPS executable:

-echo none/screen/log/both  : echoing of input script (-e)
-help                       : print this help message (-h)
-in none/filename           : read input from file or stdin (default) (-i)
-kokkos on/off ...          : turn KOKKOS mode on or off (-k)
-log none/filename          : where to send log output (-l)
-mdi '<mdi flags>'          : pass flags to the MolSSI Driver Interface
-mpicolor color             : which exe in a multi-exe mpirun cmd (-m)
-cite                       : select citation reminder style (-c)
-nocite                     : disable citation reminder (-nc)
-nonbuf                     : disable screen/logfile buffering (-nb)
-package style ...          : invoke package command (-pk)
-partition size1 size2 ...  : assign partition sizes (-p)
-plog basename              : basename for partition logs (-pl)
-pscreen basename           : basename for partition screens (-ps)
-restart2data rfile dfile ... : convert restart to data file (-r2data)
-restart2dump rfile dgroup dstyle dfile ...
                            : convert restart to dump file (-r2dump)
-reorder topology-specs     : processor reordering (-r)
-screen none/filename       : where to send screen output (-sc)
-skiprun                    : skip loops in run and minimize (-sr)
-suffix gpu/intel/opt/omp   : style suffix to apply (-sf)

**Comment by njzjz** [comment]
Thanks for reporting the problem. It got fixed in https://github.com/deepmd-kit-recipes/libdeepmd-feedstock/pull/50.

I confirm that the plugin is now loaded correctly, as shown in https://nbviewer.org/gist/njzjz/1cce3465e73a1c7055fb99582b563aac.
